src/services/services/services.service.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable, ServiceUnavailableException } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+
+import { Model, Types } from 'mongoose';
+
+import { ServiceCategoryModel, ServiceEntityModel } from '../models/entities/service.entity.model';
+import { DB_ERRORS } from '../utils/constants';
+
+@Injectable()
+export class ServicesService {
+    constructor(
+        @InjectModel('Service')
+        private readonly serviceModel: Model<ServiceEntityModel>) { }
+
+    public async createService(
+        serviceEntity: ServiceEntityModel,
+        userId: string): Promise<ServiceEntityModel> {
+
+        try {
+            serviceEntity.userId = userId;
+            const createdEntity = await this.serviceModel.create(serviceEntity);
+            return createdEntity;
+        } catch (e) {
+            throw new ServiceUnavailableException(DB_ERRORS.OfflineError);
+        }
+    }
+
+    public async getAllService(userId: string): Promise<ServiceCategoryModel[]> {
+        try {
+            const userServices = await this.serviceModel.aggregate([{
+                $lookup: {
+                    from: 'categories',
+                    localField: 'categoryId',
+                    foreignField: '_id',
+                    as: 'Categories',
+                },
+            },
+            { $match: { userId, isArchived: false } },
+            ]);
+
+            return userServices;
+        } catch (e) {
+            throw new ServiceUnavailableException(DB_ERRORS.OfflineError);
+        }
+    }
+
+    public async updateService(serviceId: string, userId: string, entityToUpdate: ServiceEntityModel): Promise<ServiceEntityModel> {
+        try {
+            const updatedEntity = await this.serviceModel.findByIdAndUpdate(serviceId, entityToUpdate).exec();
+            return updatedEntity;
+        } catch (e) {
+            throw new ServiceUnavailableException(DB_ERRORS.OfflineError);
+        }
+    }
+
+    public async deleteService(serviceId: string): Promise<void> {
+        try {
+            const service = await this.serviceModel.findById(serviceId).exec();
```

Comment:
naming is not consistent.. in some methods the variable is called as entity.. here it's called as service - both are fine, but it should be consistent.  Better to call as entity (eg.. createdEntity, updatedEntity)

Reply:
applied